pkg/scheduler/framework/plugins/volumerestrictions/volume_restrictions.go

```diff
@@ -159,7 +159,7 @@ func (pl *VolumeRestrictions) isReadWriteOncePodAccessModeConflict(pod *v1.Pod)
 		pvcKeys = append(pvcKeys, key)
 	}
 
-	ctx, cancel := context.WithCancel(context.Background())
+	subCtx, cancel := context.WithCancel(ctx)
```

Comment:
do you still need to create a context in here or can you just use the one that's passed to the function now?

Reply:
the sub-context `subCtx` try to cancel the function when "node has pod using PersistentVolumeClaim with the same name and ReadWriteOncePod access mode", and `ctx` from the function args try to cancel the function When the external cancel function is triggered. I think they are different.

Reply:
So if we want to cancel the `PreFilter` phase, why we still want to run `processNode`? I think this is the concern Mike cares?

Reply:
@denkensk @kerthcet  `subCtx` is created by `context.WithCancel`, so when parent context(`ctx` in function args)'s cancel function is called, the `processNode` function will cancel. This process (new) is working fine and this is what I added in this pr. But here is another process (old), concurrent execution of multiple `processNode ` functions, if a certain one does not meet the condition`refCount > 0`, then all running `processNode ` functions are cancelled, by `atomic.AddUint32(&conflicts, 1); cancel()`. Similar process appears in the k8s code [scheduler.preemption](https://github.com/kubernetes/kubernetes/blob/master/pkg/scheduler/framework/preemption/preemption.go#L551)

Reply:
Sorry for the rough first review, this makes sense as we may cancel the parallel program with `cancel()`. Thanks for explaining.